### PR TITLE
Do not try to render failed responses

### PR DESCRIPTION
**Pull request by blueyed**
There might be a 406 response because of a missing version header.

Not sure if this is the right approach here, but this resulted in a "406
…" message in SwaggerUI instead of a error-500 html dump.

Related: https://github.com/core-api/python-openapi-codec/issues/8

### CODE REVIEW

**Comment by blueyed** [comment]
The missing version comes via DRF's AcceptHeaderVersioning (without a default and therefore enforced version).  Do you have a pointer how this should be configured in django-rest-swagger 2.x?

**Comment by marcgibbons** [comment]
Good call.

File: rest_framework_swagger/renderers.py [hunk 1]
```diff
@@ -13,9 +14,11 @@ class OpenAPIRenderer(BaseRenderer):
     format = 'openapi'
 
     def render(self, data, accepted_media_type=None, renderer_context=None):
+        if ('response' in renderer_context and
+                ['response'].status_code != status.HTTP_200_OK):
```
Comment (ticosax) [inline rest_framework_swagger/renderers.py]: Shouldn't it be `renderer_context['response']` instead of `['response']` ?

Comment (blueyed) [inline rest_framework_swagger/renderers.py]: Of course!  Thanks!

**Comment by blueyed** [comment]
@marcgibbons 
There should be a test of course, but I am not sure if this is the best approach for this issue anyway: it would be nice to return more information / the actual error and detail of the 406 somehow - but that might get done implicitly for an empty return upwards in the stack.  I don't remember the details right now.
What do you think?

**Comment by marcgibbons** [comment]
@blueyed I've pulled your branch into #522 and added some additional handling. Thank you for this!

**Pull request closed without merging**